### PR TITLE
Feature - pil training on submission

### PR DESCRIPTION
**Pull request by joefitter**
* Added new hook to attach training an exemptions to task on submission and resubmission

### CODE REVIEW

File: lib/hooks/submission-meta/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  return model => {
+    const type = get(model, 'data.model');
+    const action = get(model, 'data.action');
+
+    if (type !== 'pil') {
+      return Promise.resolve();
+    }
+
+    if (action !== 'grant' && action !== 'transfer') {
```
Comment (lennym) [inline lib/hooks/submission-meta/index.js:14]: What are the things we're trying to ignore here? All I can think of is reviews, and it seems like perhaps they should have training attached too.

File: test/data/default.js [hunk 2]
```diff
@@ -86,7 +86,23 @@ module.exports = models => {
...
+              modules: [
+                {
+                  module: 'E1'
+                }
+              ]
+            }
+          ],
+          exemptions: [
+            {
+              id: ids.model.exemption.colinJackson,
```
Comment (lennym) [inline test/data/default.js:102]: 🤣

File: lib/hooks/submission-meta/index.js (hunk 1 above)
Comment (joefitter) [reply lib/hooks/submission-meta/index.js:14]: yeah, I thought that, then I noticed we don't show training for reviews in task views, I'm not sure _why_, perhaps for this exact reason. maybe we'll never know.

Comment (lennym) [reply lib/hooks/submission-meta/index.js:14]: I feel like it batter to have the data and not need it, than to need it and not have it, so I'd probably dispense with this check and add the metadata to all PIL related tasks.

Comment (lennym) [reply lib/hooks/submission-meta/index.js:14]: If there are some we _know_ we don't need (create?) then potentially filter those out.

**Pull request merged**